src/diffusers/pipelines/hunyuan_image/pipeline_hunyuanimage.py

```diff
@@ -0,0 +1,902 @@
+                torch.tensor(
+                    [guider.distilled_guidance_scale] * latents.shape[0], dtype=self.transformer.dtype, device=device
+                )
+                * 1000.0
+            )
+
+        else:
+            guidance = None
+
+        if self.attention_kwargs is None:
+            self._attention_kwargs = {}
+
+        # 6. Denoising loop
+        self.scheduler.set_begin_index(0)
+        with self.progress_bar(total=num_inference_steps) as progress_bar:
+            for i, t in enumerate(timesteps):
+                if self.interrupt:
+                    continue
+
+                self._current_timestep = t
+                # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
+                timestep = t.expand(latents.shape[0]).to(latents.dtype)
+
+                if self.transformer.config.use_meanflow:
+                    if i == len(timesteps) - 1:
+                        timestep_r = torch.tensor([0.0], device=device)
+                    else:
+                        timestep_r = timesteps[i + 1]
+                    timestep_r = timestep_r.expand(latents.shape[0]).to(latents.dtype)
+                else:
+                    timestep_r = None
+
+                # Step 1: Collect model inputs needed for the guidance method
+                # The `_guider_input_fields` defines which inputs model needs for conditional/unconditional predictions.
+                # e.g. {"encoder_hidden_states": ("prompt_embeds", "negative_prompt_embeds")}
+                # means both prompt_embeds (conditional) and negative_prompt_embeds (unconditional) as inputs.
+                guider_inputs = {}
+                for _, input_names_tuple in self._guider_input_fields.items():
+                    for input_name in input_names_tuple:
+                        guider_inputs[input_name] = locals()[input_name]
+
+                # Step 2: Update guider's internal state for this denoising step
+                guider.set_state(step=i, num_inference_steps=num_inference_steps, timestep=t)
+
+                # Step 3: Prepare batched model inputs based on the guidance method
+                # The guider splits model inputs into separate batches for conditional/unconditional predictions.
+                # For CFG with _guider_input_fields = {"encoder_hidden_states": ("prompt_embeds", "negative_prompt_embeds")}:
+                #   guider_state = [
+                #       {"encoder_hidden_states": prompt_embeds, "__guidance_identifier__": "pred_cond"},      # conditional batch
+                #       {"encoder_hidden_states": negative_prompt_embeds, "__guidance_identifier__": "pred_uncond"},  # unconditional batch
+                #   ]
+                # Other guidance methods may return 1 batch (no guidance) or 3+ batches (e.g., PAG, APG).
+                guider_state = guider.prepare_inputs(guider_inputs, self._guider_input_fields)
+                # Step 4: Run the denoiser for each batch
```

Comment:
(nit): When we batch the conditional and unconditional inputs together, there might be non-trivial gains in speed. We could try to allow the users to configure this in the respective guider class (doesn't have to happen in this PR, of course).